scripts/globals/regimes.lua

```diff
@@ -1050,21 +1057,27 @@ tpz.regime.bookOnEventFinish = function(player, option, regimeType)
     local msgOffset = zones[zoneId].text.REGIME_REGISTERED
     local tabs = player:getCurrency("valor_point")
     local regimeRepeat = bit.band(option, 0x80000000)
+    local hasKI  = player:hasKeyItem(tpz.ki.RHAPSODY_IN_WHITE)
 
     option = bit.band(option, 0x7FFFFFFF)
 
     -- check valid option
-    local opts = getFinishOpts(regimeType)
+    local opts = getFinishOpts(regimeType, player)
     local opt = opts[option]
 
     if not opt then
         return
     end
 
     -- check player has enough tabs
-    if opt.cost and opt.cost > tabs then
-        player:showText(player, msgOffset + 1032) -- You do not have enough tabs.
-        return
+    if hasKI then
+        if opt.discounted and opt.discounted > tabs then
+            player:showText(player, msgOffset + 1032) -- You do not have enough tabs.
+            return
+        elseif opt.cost and opt.cost > tabs then
+            player:showText(player, msgOffset + 1032) -- You do not have enough tabs.
+            return
+        end
     end
```

Comment:
Should this be checking regular cost if no KI? Like this:
```lua
    if hasKI then
        if opt.discounted and opt.discounted > tabs then
            player:showText(player, msgOffset + 1032) -- You do not have enough tabs.
            return
        end
    else
        if opt.cost and opt.cost > tabs then
            player:showText(player, msgOffset + 1032) -- You do not have enough tabs.
            return
        end
    end
```

Reply:
fixed :)